client/cs/src/ObjectHandle.cs

```diff
@@ -8,7 +8,7 @@ namespace Teleportal.Client.Object
 {
     public sealed class ObjectHandle : OpaqueWrapper<ObjectHandle>
     {
-        public ObjectHandle(Ptr<ObjectHandle> inner) : base(inner, OwnershipSemantics.Owned) { }
+        public ObjectHandle(Ptr<ObjectHandle> inner) : base(inner, OwnershipSemantics.SharedRef) { }
```

Comment:
I don't think this is correct - all our object handles are owned, aren't they? Certainly *most* of them are owned.

If there are any un-owned object handles, can you link me?

Reply:
Looks like this is because we switched all of the handle types to use references.

Originally we had no `ownershipSemantics` parameter because things were only owned. Now, mostly they are references - but I don't want to make the assumption that it is only one or the other anymore.

Instead, our constructor should have an `ownershipSemantics` parameter same as the majority of the other wrapper types, and the movieoke code should be explicitly passing whether its owned or shared when it constructs the wrapper.

Reply:
Updating. Should we enforce shared reference semantics in future for handles? In which case(s) would owned semantics be necessary for handles?

Reply:
restricting C# to only shared ref semantics implies that the rust side must always own the data. But logically this is not possible, because C# wants to hang onto handles for longer than just a particular rust lifetime. For example, when we create an object, we currently give to C# an owned `ObjectHandle` (i.e. repr_c::Box<ObjectHandle>`). 

So we need to support both shared and owned semantics in C#, hence this needs the `ownershipSemantics` parameter as part of the constructor